Sources/Rendering/Core/Renderer/index.js

```diff
@@ -464,6 +464,149 @@ function vtkRenderer(publicAPI, model) {
+          ? model._renderWindow.getViews()
+          : [];
+        if (views.length > 0) {
+          const view = views[0];
+          const dims = view.getViewportSize
+            ? view.getViewportSize(publicAPI)
+            : null;
+          if (dims && dims[0] > 0 && dims[1] > 0) {
+            const aspect = dims[0] / dims[1];
+
+            // Get corner points of the bounds in world space
+            const visiblePoints = [];
+            vtkBoundingBox.getCorners(boundsToUse, visiblePoints);
+
+            // Transform bounds to view space using the view matrix
+            // The view matrix is now valid since we've set up the camera
+            const viewBounds = vtkBoundingBox.reset([]);
+            const viewMatrix = model.activeCamera.getViewMatrix();
+            const viewMatrixTransposed = new Float64Array(16);
+            mat4.copy(viewMatrixTransposed, viewMatrix);
+            mat4.transpose(viewMatrixTransposed, viewMatrixTransposed);
+
+            for (let i = 0; i < visiblePoints.length; ++i) {
+              const point = visiblePoints[i];
+              const viewPoint = new Float64Array(3);
+              vec3.transformMat4(viewPoint, point, viewMatrixTransposed);
+              vtkBoundingBox.addPoint(viewBounds, ...viewPoint);
+            }
+
+            // Get lengths in view space
+            const xLength = vtkBoundingBox.getLength(viewBounds, 0);
+            const yLength = vtkBoundingBox.getLength(viewBounds, 1);
+
+            // Apply offset ratio to add white space buffer
+            // offsetRatio is the fraction of space to use (default 0.9 = 90%, leaving 10% margin)
+            const marginMultiplier = 1.0 / offsetRatio;
+            const xLengthWithMargin = marginMultiplier * xLength;
+            const yLengthWithMargin = marginMultiplier * yLength;
+
+            // Use max of height and width/aspect to ensure everything fits
+            // This accounts for viewport aspect ratio to prevent cropping
+            // This mirrors C++ VTK behavior
+            parallelScale =
+              0.5 * Math.max(yLengthWithMargin, xLengthWithMargin / aspect);
+          }
+        }
+      } catch (e) {
+        // If we can't get aspect ratio, fall back to using radius
+        vtkDebugMacro(
+          'ResetCameraScreenSpace could not get aspect ratio, using radius for parallel scale'
+        );
+      }
+    }
+
+    publicAPI.resetCameraClippingRange(boundsToUse);
+
+    // setup parallel scale (computed from view space for parallel projection)
+    model.activeCamera.setParallelScale(parallelScale);
+
+    // update reasonable world to physical values
+    model.activeCamera.setPhysicalScale(radius);
+    model.activeCamera.setPhysicalTranslation(
+      -center[0],
+      -center[1],
+      -center[2]
+    );
+
+    // Here to let parallel/distributed compositing intercept
+    // and do the right thing.
+    publicAPI.invokeEvent(RESET_CAMERA_EVENT);
+
+    return true;
+  };
+
```

Comment:
why not recoding the [VTK C++ implementation](https://gitlab.kitware.com/vtk/vtk/-/blob/master/Rendering/Core/vtkRenderer.cxx?ref_type=heads#L1306) ? 

This ensures you have the same behavior between VTK C++ and VTK.js. Which is important.